burgerSync.js

```diff
@@ -0,0 +1,252 @@
+        }
+    }
+
+    /**
+     * Broadcasts a new transaction to all connected peers.
+     * 
+     * @param {burgerTransaction} transaction 
+     */
+    broadcastNewTransaction(transaction) {
+        this._broadcast(this.MESSAGE_TYPE.BROADCAST_NEW_TRANSACTION, transaction);
+    }
+
+    /**
+     * Broadcasts a new block to all connected peers.
+     * This event happens when a new block is mined.
+     * 
+     * @param {burgerBlockchain} chain 
+     */
+    broadcastNewBlock(chain) {
+        this._broadcast(this.MESSAGE_TYPE.BROADCAST_NEW_BLOCK, chain);
+    }
+
+    /**
+     * Removes the peer from the (nodeId => websocket) map.
+     * 
+     * @param {websocket} peer - the socket connection object to the peer
+     */
+    initializeErrorHandler(peer) {
+        this._printMessage('Connection Failed to ' + this._getInternalQualifiedName(peer));
+        this._removePeer(peer);
+    }
+
+    // Private methods ================
+
+    /**
+     * Sends a message to a single peer.
+     * 
+     * @param {websocket} peer - the socket connection object to the peer.
+     * @param {burgerSync.MESSAGE_TYPE} type - the message type (from burgerSync.MESSAGE_TYPE).
+     * @param {object} message - the message to be sent
+     */
+    _write(peer, type, message) {
+        const transmission = { type, message };
+        peer.send(JSON.stringify(transmission));
+    }
+
+    /**
+     * Broadcasts a message to all connected peers.
+     * 
+     * @param {burgerSync.MESSAGE_TYPE} type - the message type (from burgerSync.MESSAGE_TYPE).
+     * @param {object} message - the message to be sent
+     */
+    _broadcast(type, message) {
+        this._getPeers().forEach(function(peer) {
+            this._write(peer, type, message);
+        }.bind(this));
+    }
+
+    /**
+     * Prints a log on the console.
+     * 
+     * @param {string} message 
+     */
+    _printMessage(message) {
+        console.log(`[${new Date().toISOString()}] ${message}`);
+    }
+
+    /**
+     * Gets the [remoteAddress:port] string of a websocket connection
+     * object for whatever purpose it may serve us best. :D
+     * 
+     * @param {*} peer 
+     */
+    _getInternalQualifiedName(peer) {
+        return peer._socket.remoteAddress + ":" + peer._socket.remotePort;
+    }
+
+    /**
+     * Removes a peer from the (nodeId => websocket) map.
+     * 
+     * @param {url} peerUrl - the peer URL (a string, not the websocket instance)
+     */
+    _removePeer(peerUrl) {
+        const peerIndex = Object.values(this.peers).indexOf(peerUrl);
+        const peer = Object.keys(this.peers)[peerIndex];
+        delete this.peers[peer];
+        delete this.burgerNode.nodes[peer];       
+    }
+
+    /**
+     * Retrieves all websocket instances from the (nodeId => websocket) map.
+     */
+    _getPeers() {
+        return Object.values(this.peers);
+    }
+}
+
+module.exports = BurgerSync;
```

Comment:
@24thsaint just a note, git will always complain if there is no newline at the end of the file.

Reply:
Noted.